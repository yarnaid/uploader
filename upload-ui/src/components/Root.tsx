import React, { useEffect } from "react";
import { Routes, Route, useLocation } from "react-router-dom";
import { UploadNavbar } from "./UploadNavbar";
import FilePage from "./pages/FilePage";
import HomePage from "./pages/HomePage";
import { useDispatch } from "react-redux";
import { resetState as resetFilesState } from "../redux/reducers/files";
import { resetState as resetFileState } from "../redux/reducers/file";
import Filters from "./pages/EditFiltersPage";
import { loadAllFilters } from "../redux/reducers/filters";
import { resetState as resetTagsState } from "../redux/reducers/tags";

function Root() {
  const dispatch = useDispatch();

  const location = useLocation();
  useEffect(() => {
    dispatch(resetFilesState());
    dispatch(resetFileState());
    dispatch(resetTagsState());
    dispatch(loadAllFilters());
  }, [dispatch, location]);

  return (
    <>
      <UploadNavbar />
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/files" element={<HomePage />} />
        <Route path="/files/:fileId" element={<FilePage />} />
        <Route path="/upload" element={<FilePage />} />
        <Route path="/edit-filters" element={<Filters />} />
      </Routes>
    </>
  );
}

export default Root;
