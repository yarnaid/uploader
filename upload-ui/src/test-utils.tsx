import React, { ReactElement } from "react";
import { Provider } from "react-redux";
import { render, RenderOptions } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "./redux/reducers";
import {
  FileMetadataType,
  FilterType,
  FileType,
  UploadFileType,
} from "./types";
import type { PreloadedState } from "@reduxjs/toolkit";
import type { RootState, AppStore } from "./redux/store";

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const mockStore = configureMockStore();

const initialState: {
  files: { files: FileType[] };
  filters: FilterType[];
  file: string;
  tags: FilterType[];
  metadata: FileMetadataType;
} = {
  files: { files: [] },
  filters: [],
  file: "",
  tags: [],
  metadata: { tags: [] },
};

export const store: any = mockStore(initialState);

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => render(ui, { wrapper: AllTheProviders, ...options });
export * from "@testing-library/react";

export { customRender as render };
