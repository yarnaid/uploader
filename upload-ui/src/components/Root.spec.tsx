import React from "react";
import Root from "./Root";
import { render } from "../test-utils";
import { screen } from "@testing-library/react";

describe("Root", () => {
  it("renders", () => {
    const { asFragment } = render(<Root />);
    expect(screen.getAllByText(/upload/i)[0]).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
