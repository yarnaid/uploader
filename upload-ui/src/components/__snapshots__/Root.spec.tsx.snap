// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Root > renders 1`] = `
<DocumentFragment>
  <nav
    class="bg-body-tertiary navbar navbar-expand navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="#home"
      >
        Upload Service
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="me-auto my-2 my-lg-0 navbar-nav"
      >
        <a
          class="nav-link"
          href="/"
        >
          Home
        </a>
        <a
          class="nav-link"
          href="/upload"
        >
          Upload
        </a>
        <a
          class="nav-link"
          href="/edit-filters"
        >
          Edit Metadata Settings
        </a>
      </div>
    </div>
  </nav>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h1>
              Files
            </h1>
            <table
              class="table"
            >
              <thead>
                <tr>
                  <th>
                    <span>
                      Name
                    </span>
                  </th>
                  <th>
                    Updated
                  </th>
                  <th>
                    Created
                  </th>
                  <th>
                    Size
                  </th>
                </tr>
              </thead>
              <tbody />
              <tfoot>
                <tr>
                  <th>
                    name
                  </th>
                  <th>
                    updated
                  </th>
                  <th>
                    created
                  </th>
                  <th>
                    size
                  </th>
                </tr>
              </tfoot>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root renders 1`] = `
<DocumentFragment>
  <nav
    class="bg-body-tertiary navbar navbar-expand navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="#home"
      >
        Upload
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="me-auto my-2 my-lg-0 navbar-nav"
      >
        <a
          class="nav-link"
          href="/"
        >
          Home
        </a>
        <a
          class="nav-link"
          href="/upload"
        >
          Upload
        </a>
        <a
          class="nav-link"
          href="/edit-filters"
        >
          Edit Metadata Settings
        </a>
      </div>
    </div>
  </nav>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-6"
      >
        <h1>
          Upload UI
        </h1>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h1>
              Files
            </h1>
            <table
              class="table"
            >
              <thead>
                <tr>
                  <th>
                    <span>
                      Name
                    </span>
                  </th>
                  <th>
                    Updated
                  </th>
                  <th>
                    Created
                  </th>
                  <th>
                    Size
                  </th>
                </tr>
              </thead>
              <tbody />
              <tfoot>
                <tr>
                  <th>
                    name
                  </th>
                  <th>
                    updated
                  </th>
                  <th>
                    created
                  </th>
                  <th>
                    size
                  </th>
                </tr>
              </tfoot>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
